MergerService/src/run.cs

```diff
@@ -321,8 +325,8 @@ private void RunTask(MergeTask task, ITaskUtils taskUtils, string? managerCallba
                         IData target = sources[0];
                         target.IsNew = metadata.IsNewTarget;
 
-                        // TODO: fix to use inner batch size (add iteration inside loop below)
-                        List<Tile> tiles = new List<Tile>((int)singleTileBatchCount);
+                        List<Tile> tiles = new List<Tile>(this._batchSize);
```

Comment:
Should only be defined with _batchSize if _limitBatchSize is set to true

Reply:
done